spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/ConcurrentKafkaListenerContainerFactoryConfigurer.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2012-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.kafka;
+
+import org.springframework.boot.autoconfigure.kafka.KafkaProperties.Listener;
+import org.springframework.kafka.config.ConcurrentKafkaListenerContainerFactory;
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.kafka.listener.config.ContainerProperties;
+
+/**
+ * Default configurer for Kafka listener container factories.
+ *
+ * @author Gary Russell
+ * @since 1.5
+ *
+ */
+public class ConcurrentKafkaListenerContainerFactoryConfigurer {
+
+	private KafkaProperties properties;
+
+	/**
+	 * Set the {@link KafkaProperties} to use.
+	 * @param properties the properties
+	 */
+	public void setKafkaProperties(KafkaProperties properties) {
+		this.properties = properties;
+	}
```

Comment:
Is there a reason why this is not set via constructor injection? If you make the link with the JMS part we had a bunch of components that might be responsible of tuning the factory and we didn't want to offer a way to users to modify the configurer once it has been created. That's why they are package protected (this one is public).